core/src/main/java/org/apache/carbondata/core/datastore/TableSpec.java

```diff
@@ -103,7 +102,7 @@ private void addDimensions(List<CarbonDimension> dimensions) {
         } else {
           noSortNoDictDimSpec.add(spec);
         }
-      } else if (dimension.isDirectDictionaryEncoding()) {
```

Comment:
CarbonDimension has isGlobalDictionaryEncoding and isDirectDictionaryEncoding, please remove both the methods and handle the callers